app/jobs/calendar_importer/calendar_importer_task.rb

```diff
@@ -29,53 +29,54 @@ def run
       CalendarImporter::EventResolver.new(event_data, calendar, notices, from_date)
     end
 
-    return if parsed_events.blank?
-
-    all_event_uids = Set.new(calendar.events.upcoming.pluck(:uid))
-    active_event_uids = Set.new
-
-    # this can be useful if you want to force import on your local machine
-    # if @force_import
-    #   calendar.events.destroy_all
-    # end
-
-    parsed_events.each do |parsed_event|
-      # is source event valid?
-      # now find all occurrences of event in calendar
-      # set up location from strategy
-
-      # find all events in calendar with same UID
-      # if recurring event?
-      #   delete upcoming calendar events that have the same start and end time as our parsed_event
-      #
-      # for each occurence
-      #   skip if more than a day apart (?)
-      #   setup start_time and end_time
-      #   if this is recurring
-      #     try to find an event with the same start time and end time
-      #     or make a new event
-      #   set are_spaces_available
-      #   save event
-      #
-      # delete all upcoming events that have not been created/updated by this import
-
-      next if parsed_event.is_private?
-      next if parsed_event.has_no_occurences?
-
-      parsed_event.determine_online_location
-
-      parsed_event.determine_location_for_strategy
-      # next if parsed_event.is_address_missing?
-
-      active_event_uids << parsed_event.uid
-
-      parsed_event.save_all_occurences
-
-    rescue CalendarImporter::EventResolver::Problem => e
-      notices << e.message
-    end
+    if parsed_events.present?
```

Comment:
This is a very big method could it be broken up a bit to make it easier to read what's going on? Maybe the big if loop in a helper method?

Reply:
I agree it is a touch long but 30% of it is comment explaining the algorithm. Could be moved into helper methods but would that improve parsability? Much of the fiddly logic from the original has been moved into the EventResolver helper class already.

Reply:
i think the thing that increases parsability is simply having a method named say, "make_events_do_good_thing" so you can then just read the method name and not to get into the weeds. if that makes sense! almost kinda like writing in pseudocode except its not pseudocode?

Reply:
like yes it might increase the overall LoC but just having a simple human parsable top level method makes it a ton easier to remember whats going on i think

Reply:
i believe you are describing "self documenting code". i will have a stab at it.

yeah rubocop is way too strict out of the box but its better to be too strict and loosen up than have nothing and add rules (like- otherwise why bother?)